portfolio/src/main/webapp/script.js

```diff
@@ -12,6 +12,39 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+/** Uses fetch() to add a message to the DOM */
+ function getMessage() {
+    // fetches from the DataServlet
+    const responsePromise = fetch("/data");
+
+    // passes the response to handleResponse after the request is complete
+    responsePromise.then(handleResponse);
+ }
+
+/**
+ * Handles response by converting it to text and passing the result to
+ * addQuoteToDom().
+ */
+ function handleResponse(response) {
+    const textPromise = response.text();
+
+    textPromise.then(addMessageToDom);
+ }
+
+/** Adds the message to the DOM */
+function addMessageToDom(message) {
+
+    const messageContainer = document.getElementById('message_container');
+    messageContainer.innerText = message;
+}
+
+/** Practices using arrow functions to shorten the code */
```

Comment:
Lambda functions (like you're using here) are super common in JS and a nice way to condense some simple code.
Nice! 👍